File: html/en/7gui/counter.c
#include <stdlib.h>
#include <iup.h>

int btn_count_cb( Ihandle *self )
{
  Ihandle* text = IupGetDialogChild(self, "TEXT");
  int value = IupGetInt(text, "VALUE");
  IupSetInt(text, "VALUE", ++value);

  return IUP_DEFAULT;
}

int main(int argc, char **argv)
{
  Ihandle *dlg, *button, *text, *hbox;

  IupOpen(&argc, &argv);
  
  button = IupButton("Count", NULL);
  IupSetAttribute(button, "SIZE", "60");
  text = IupText(NULL);
  IupSetAttribute(text, "SIZE", "60");
  IupSetAttribute(text, "NAME", "TEXT");
  IupSetAttribute(text, "READONLY", "YES");
  hbox = IupHbox(
    text,
    button,
    NULL);
  IupSetAttribute(hbox, "MARGIN", "10x10");
  IupSetAttribute(hbox, "GAP", "10");
  dlg = IupDialog(hbox);
  IupSetAttribute(dlg, "TITLE", "Counter");

  IupSetInt(text, "VALUE", 0);

  /* Registers callbacks */
  IupSetCallback(button, "ACTION", (Icallback) btn_count_cb);

  IupShowXY(dlg, IUP_CENTER, IUP_CENTER);

  IupMainLoop();

  IupClose();
  return EXIT_SUCCESS;
}

File: html/en/7gui/counter.lua
require("iuplua")

counter = 0

function addCount()
	counter = counter + 1
end

function getCount()
	return counter
end

--********************************** Main *****************************************

txt_count = iup.text{value = getCount(), readonly = "YES",  size = "60"}
btn_count = iup.button{title = "Count", size = "60"}

dlg = iup.dialog{iup.hbox{txt_count, btn_count; ngap = "10"}, title = "Counter", margin = "10x10"}

function btn_count:action()
  addCount()
  txt_count.value = getCount()
end

dlg:showxy( iup.CENTER, iup.CENTER )

if (iup.MainLoopLevel()==0) then
  iup.MainLoop()
end
